### DISCUSSION TITLE
Add Sky Programmatically

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#1955] [Support Q&A] [answered]
How can I add sky with sun shading effect programmatically?

Thanks

Edit: I could add sky programmatically. But I could not show the black background, I mean the space.

**Comment by gwaldron** [comment, marked as answer]
```c++
        auto sky = SkyNode::create();
        sky->attach(&viewer);
        root->addChild(sky);
```
Then add your `MapNode` under the sky.

**Reply by nebsar** [reply]
Thanks Glenn. I did not attache the sky to viewer before. That was what I missed.
